File: codegen/results/v002/m3-max-48gb/lmstudio/qwen3.8-27b-mlx-4bit_think/20260830_233604/typescript/007_beatles_interview/solution.ts
import * as fs from 'fs';
import * as path from 'path';

const inputPath = path.join('input', 'input.csv');
const content = fs.readFileSync(inputPath, 'utf-8');

const lines = content.trim().split('\n');
const headers = lines[0].split(',').map(h => h.trim());

interface Relative {
  FirstName: string;
  LastName: string;
  Relationship: string;
}

interface Person {
  FirstName: string;
  LastName: string;
  Birthday: string;
  Age: number;
  Relatives: Relative[];
}

function parseName(fullName: string): { first: string; last: string } {
  const parts = fullName.trim().split(' ');
  const first = parts[0];
  const last = parts[parts.length - 1];
  return { first, last };
}

function parseDate(dateStr: string): { month: number; day: number; year: number } {
  const parts = dateStr.split('/').map(Number);
  return { month: parts[0], day: parts[1], year: parts[2] };
}

function toISO(month: number, day: number, year: number): string {
  return `${year}-${String(month).padStart(2, '0')}-${String(day).padStart(2, '0')}`;
}

function calculateAge(birthDate: string, deathDate: string | null, refDate: Date): number {
  const ref = deathDate
    ? parseDate(deathDate)
    : { month: refDate.getMonth() + 1, day: refDate.getDate(), year: refDate.getFullYear() };

  const birth = parseDate(birthDate);

  let age = ref.year - birth.year;
  if (ref.month < birth.month || (ref.month === birth.month && ref.day < birth.day)) {
    age--;
  }

  return age;
}

const referenceDate = new Date(2025, 6, 1); // July 1, 2025

const result: Person[] = [];

for (let i = 1; i < lines.length; i++) {
  const values = lines[i].split(',').map(v => v.trim());
  const row: Record<string, string> = {};
  headers.forEach((h, idx) => {
    row[h] = values[idx] ?? '';
  });

  const nameParts = parseName(row['Name']);

  const birthday = parseDate(row['Birthday']);
  const birthdayStr = toISO(birthday.month, birthday.day, birthday.year);

  const died = row['Died'] === 'null' || row['Died'] === '' ? null : row['Died'];

  const age = calculateAge(row['Birthday'], died, referenceDate);

  const relatives: Relative[] = [];
  const relColumns = ['Father', 'Mother', 'Brother', 'Sister'];

  for (const col of relColumns) {
    const val = row[col];
    if (val && val !== 'null') {
      const relName = parseName(val);
      relatives.push({
        FirstName: relName.first,
        LastName: relName.last,
        Relationship: col
      });
    }
  }

  result.push({
    FirstName: nameParts.first,
    LastName: nameParts.last,
    Birthday: birthdayStr,
    Age: age,
    Relatives: relatives
  });
}

process.stdout.write(JSON.stringify(result, null, 2) + '\n');

File: codegen/results/v002/m3-max-48gb/lmstudio/qwen3.8-27b-mlx-4bit_think/20260830_233604/python/007_beatles_interview/solution.py
import csv
import json
from datetime import date

def parse_date(s):
    """Parse a date string like '10/9/1940' into a date object."""
    parts = s.strip().split('/')
    return date(int(parts[2]), int(parts[0]), int(parts[1]))

def format_date(d):
    """Format a date object as YYYY-MM-DD."""
    return d.strftime('%Y-%m-%d')

def calc_age(birthday, reference_date):
    """Calculate age at reference_date given a birthday."""
    age = reference_date.year - birthday.year
    if (reference_date.month, reference_date.day) < (birthday.month, birthday.day):
        age -= 1
    return age

def parse_name(full_name):
    """Split 'First [Middle] Last' into (FirstName, LastName)."""
    parts = full_name.strip().split()
    first_name = parts[0]
    last_name = parts[-1]
    return first_name, last_name

def parse_relative(name, relationship):
    """Parse a relative's name and relationship into a dict, or None."""
    if name is None or name.strip().lower() == 'null' or name.strip() == '':
        return None
    parts = name.strip().split()
    first_name = parts[0]
    last_name = parts[-1]
    return {
        "FirstName": first_name,
        "LastName": last_name,
        "Relationship": relationship
    }

reference_date = date(2025, 7, 1)

results = []

with open('input/input.csv', 'r') as f:
    reader = csv.DictReader(f)
    for row in reader:
        first_name, last_name = parse_name(row['Name'])
        birthday = parse_date(row['Birthday'])

        # Determine reference date for age: death date if exists, else July 1, 2025
        died_str = row['Died'].strip()
        if died_str.lower() == 'null' or died_str == '':
            ref = reference_date
        else:
            ref = parse_date(died_str)

        age = calc_age(birthday, ref)

        # Build relatives list
        relatives = []
        rel_fields = [
            ('Father', 'Father'),
            ('Mother', 'Mother'),
            ('Brother', 'Brother'),
            ('Sister', 'Sister'),
        ]
        for col, rel_name in rel_fields:
            r = parse_relative(row[col], rel_name)
            if r is not None:
                relatives.append(r)

        results.append({
            "FirstName": first_name,
            "LastName": last_name,
            "Birthday": format_date(birthday),
            "Age": age,
            "Relatives": relatives
        })

print(json.dumps(results, indent=2))